webapp/src/components/CurrencyTable.jsx

```diff
@@ -29,33 +31,72 @@ export const CurrencyTable = () => {
   // Updates selected currency in global state and closes modal
   function handleCurrencyChange(newTokenIndex) {
 
-    // Find URL of token logo, symbol, and type
+    // Find URL of token logo, symbol, type, and address
     let newImageUrl = findCurrencyLogoUrl(newTokenIndex, tokenListState.state.tokenList);
     let newSymbol = tokenListState.state.tokenList.tokens[newTokenIndex].symbol;
     let newType = tokenListState.state.tokenList.tokens[newTokenIndex].type;
+    let newAddress = tokenListState.state.tokenList.tokens[newTokenIndex].address;
 
     // Find correct input to update
     switch (inputs.state.currencySelectionModal.selectedInput) {
       case 'from':
         dispatch({ type: 'UPDATE_FROM_SELECTED_CURRENCY',
-          payload: { tokenIndex: newTokenIndex, logoUrl: newImageUrl, symbol: newSymbol, type: newType }
+          payload: { 
+            tokenIndex: newTokenIndex, 
+            logoUrl: newImageUrl, 
+            symbol: newSymbol, 
+            type: newType, 
+            address: newAddress }
         });
+
+        // Update allowance of from token
+        (async function () { 
+          if (newType == "NEP-21") {
+            await delay(1000).then(async function() {
+              try {
+                let allowance = await getAllowance(inputs.state.swap.from);
+                dispatch({ type: 'UPDATE_FROM_ALLOWANCE', payload: { allowance: allowance } });
+              } catch (e) {
+                console.error(e);
+              }
+            });
+          }
+        })();
```

Comment:
let's move this function out to a modules scope. The function can take all necessary arguments as a parameter.

In general we should avoid making closures whenever possible. It's easy to create mess and lengthy not readable functions.